Moblin/HaishinKit/Media/AudioUnit.swift

```diff
@@ -175,6 +169,39 @@ final class AudioUnit: NSObject {
     func removeReplaceAudioInner(cameraId: UUID) {
         replaceAudios.removeValue(forKey: cameraId)
     }
+
+    func prepareSampleBuffer(sampleBuffer: CMSampleBuffer, audioLevel: Float, numberOfAudioChannels: Int) {
+        guard let mixer else {
+            return
+        }
+
+        // Workaround for audio drift on iPhone 15 Pro Max running iOS 17. Probably issue on more models.
+        let presentationTimeStamp = syncTimeToVideo(mixer: mixer, sampleBuffer: sampleBuffer)
+        guard mixer.useSampleBuffer(presentationTimeStamp, mediaType: AVMediaType.audio) else {
+            return
+        }
+        mixer.delegate?.mixer(
+            audioLevel: audioLevel,
+            numberOfAudioChannels: numberOfAudioChannels,
+            presentationTimestamp: presentationTimeStamp.seconds
+        )
+        appendSampleBuffer(sampleBuffer, presentationTimeStamp, isFirstAfterAttach: false)
+    }
+
+    private func syncTimeToVideo(mixer: Mixer, sampleBuffer: CMSampleBuffer) -> CMTime {
```

Comment:
I prefer leaving this as a normal function on file level unless it needs something in the class.